src/v/storage/log_manager.cc

```diff
@@ -435,8 +435,12 @@ ss::future<> log_manager::housekeeping_loop() {
 
         try {
             co_await housekeeping_scan(lowest_ts_to_retain());
-        } catch (const std::exception& e) {
-            vlog(stlog.info, "Error processing housekeeping(): {}", e);
+        } catch (...) {
+            auto eptr = std::current_exception();
+            if (ssx::is_shutdown_exception(eptr)) {
+                std::rethrow_exception(eptr);
+            }
```

Comment:
> I was more referring to the fact that we have to iterate through everything again, rather than just continuing onto the next log. 

> moving replicas around temporarily disrupts progress of the housekeeping loop.

I don't see how this is true, `housekeeping_scan()` is the last piece of work performed in the `while` loop within `housekeeping_loop()`, and it seems to me that throwing and retrying in the `while` loop in `housekeeping()` is not functionally different than not throwing and retrying in the `while` loop in `housekeeping_loop()`.